plugins/inputs/nfsclient/nfsclient.go

```diff
@@ -0,0 +1,502 @@
+		"directwritebytes",
+		"serverreadbytes",
+		"serverwritebytes",
+		"readpages",
+		"writepages",
+	}
+
+	var xprtudpFields = []string{
+		"bind_count",
+		"rpcsends",
+		"rpcreceives",
+		"badxids",
+		"inflightsends",
+		"backlogutil",
+	}
+
+	var xprttcpFields = []string{
+		"bind_count",
+		"connect_count",
+		"connect_time",
+		"idle_time",
+		"rpcsends",
+		"rpcreceives",
+		"badxids",
+		"inflightsends",
+		"backlogutil",
+	}
+
+	var nfsopFields = []string{
+		"ops",
+		"trans",
+		"timeouts",
+		"bytes_sent",
+		"bytes_recv",
+		"queue_time",
+		"response_time",
+		"total_time",
+		"errors",
+	}
+
+	var fields = make(map[string]interface{})
+
+	switch first {
+	case "READ", "WRITE":
+		fields["ops"] = nline[0]
+		fields["retrans"] = nline[1] - nline[0]
+		fields["bytes"] = nline[3] + nline[4]
+		fields["rtt"] = nline[6]
+		fields["exe"] = nline[7]
+		tags["operation"] = first
+		acc.AddFields("nfsstat", fields, tags)
+	}
+
+	if fullstat {
+		switch first {
+		case "events":
+			if len(nline) >= len(eventsFields) {
+				for i, t := range eventsFields {
+					fields[t] = nline[i]
+				}
+				acc.AddFields("nfs_events", fields, tags)
+			}
+
+		case "bytes":
+			if len(nline) >= len(bytesFields) {
+				for i, t := range bytesFields {
+					fields[t] = nline[i]
+				}
+				acc.AddFields("nfs_bytes", fields, tags)
+			}
+
+		case "xprt":
+			if len(line) > 1 {
+				switch line[1] {
+				case "tcp":
+					if len(nline)+2 >= len(xprttcpFields) {
+						for i, t := range xprttcpFields {
+							fields[t] = nline[i+2]
+						}
+						acc.AddFields("nfs_xprt_tcp", fields, tags)
+					}
+				case "udp":
+					if len(nline)+2 >= len(xprtudpFields) {
+						for i, t := range xprtudpFields {
+							fields[t] = nline[i+2]
+						}
+						acc.AddFields("nfs_xprt_udp", fields, tags)
+					}
+				}
+			}
+		}
+
+		if (version == "3" && n.nfs3Ops[first]) || (version == "4" && n.nfs4Ops[first]) {
+			tags["operation"] = first
+			if len(nline) <= len(nfsopFields) {
+				for i, t := range nline {
+					fields[nfsopFields[i]] = t
+				}
+				acc.AddFields("nfs_ops", fields, tags)
+			}
+		}
+
+	}
+
+	return nil
+}
+
+func (n *NFSClient) processText(scanner *bufio.Scanner, acc telegraf.Accumulator) error {
+	var mount string
+	var version string
+	var export string
+	var skip bool
+
+	for scanner.Scan() {
+		line := strings.Fields(scanner.Text())
+
+		line_len := len(line)
+
+		if line_len == 0 {
+			continue
+		}
+
+		skip = false
+
+		// This denotes a new mount has been found, so set
+		// mount and export, and stop skipping (for now)
+		if line_len > 4 && choice.Contains("fstype", line) && (choice.Contains("nfs", line) || choice.Contains("nfs4", line)) {
+			mount = line[4]
+			export = line[1]
+			skip = false
+		} else if line_len > 5 && (choice.Contains("(nfs)", line) || choice.Contains("(nfs4)", line)) {
+			version = strings.Split(line[5], "/")[1]
+		}
+
+		if mount == "" {
+			continue
+		}
+
+		if len(n.IncludeMounts) > 0 {
+			skip = true
```

Comment:
Would it make sense to condense all initialization to
```suggestion
		skip = len(n.IncludeMounts) > 0
```

Reply:
You could.... But I think that it hurts code readability.  The code `if len()` makes it clear what the is being tested, and what actions come from that.  While functionally the same, and a common idiom, using assignments from comparisons hides the intent somewhat, and makes later comprehension harder (not to say "hard", just "harder").

```
-               if len(n.IncludeMounts) > 0 {
-                       skip = true
+               skip = len(n.IncludeMounts) > 0
+               if skip {
```